ui-components/src/forms/Textarea.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from "react"
+import "./Textarea.scss"
+import { t } from "../helpers/translator"
+
+type WrapOptions = "soft" | "hard"
+
+export interface TextareaProps {
+  resize?: boolean
+  cols?: number
+  rows?: number
+  wrap?: WrapOptions
+  maxLength?: number
+  placeholder?: string
+  disabled?: boolean
+  errorMessage?: string
+  label: string
+}
+
+export const Textarea = (props: TextareaProps) => {
+  const textareaClassnames = ["textarea"]
+  if (props.resize === false) textareaClassnames.push("textarea-resize-off")
+  if (props.disabled) textareaClassnames.push("textarea-disabled")
+  if (props.errorMessage) textareaClassnames.push("textarea-error")
+  const labelClassnames = ["textarea-label"]
+  if (props.errorMessage) labelClassnames.push("textarea-label-error")
+
+  return (
+    <>
+      <label className={labelClassnames.join(" ")} htmlFor={props.label}>
```

Comment:
The contents of `htmlFor` will need to match an `id` in the `textarea` tag, otherwise the association won't get made. (aka clicking on the label won't focus the textarea)